### PR TITLE
Update GAE maven plugin configs and instructions

### PR DESCRIPTION
**Pull request by averikitsch**
When the app engine maven plugin was updated it needed updates to the configs and instructions.

### CODE REVIEW

File: appengine-java11/gaeinfo/README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Main class to your classpath:
 
 - Deploy to App Engine standard environment using the following Maven command.
 ```
-   mvn appengine:deploy -Dapp.deploy.projectId=<your-project-id>
+   mvn package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java11/gaeinfo/README.md]: Can't you use GCLOUD to use the default project?  Why make the line this long?

File: appengine-java8/bigtable/README.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Then you need to [Create a Cloud Bigtable Instance](https://cloud.google.com/big
 
 ### Deploy to App Engine Standard for Java 8
 
-    mvn -Dbigtable.projectID=PROJECTID -Dbigtable.instanceID=INSTANCEID appengine:deploy
+    mvn -Dbigtable.projectID=PROJECTID -Dbigtable.instanceID=INSTANCEID package appengine:deploy
```
Comment (lesv) [inline appengine-java8/bigtable/README.md]: Why doesn't this require the -Dapp.deploy.projectId=<your-project-id>
 

 ??

File: appengine-java8/cloudsql-postgres/README.md [hunk 3]
```diff
@@ -38,12 +38,11 @@ $ mvn clean appengine:run -DINSTANCE_CONNECTION_NAME=instanceConnectionName -Dus
 ## Deploying
 
 ```bash
-$ mvn clean appengine:deploy -DINSTANCE_CONNECTION_NAME=instanceConnectionName -Duser=root
--Dpassword=myPassword -Ddatabase=myDatabase
+$ mvn clean package appengine:deploy -DINSTANCE_CONNECTION_NAME=instanceConnectionName -Duser=root
+-Dpassword=myPassword -Ddatabase=myDatabase -Dapp.deploy.projectId=projectId
 ```
```
Comment (lesv) [inline appengine-java8/cloudsql-postgres/README.md]: wow that's getting long.

File: appengine-java8/images/README.md [hunk 4]
```diff
@@ -36,4 +36,4 @@ Using the [Google Cloud SDK](https://cloud.google.com/sdk/) create a bucket
  [Google Cloud Project ID](https://developers.google.com/console/help/new/#projectnumber)
  and SOME-VERSION with a valid version number.
 
-     $ mvn appengine:update -Dappengine.appId=YOUR-PROJECT-ID -Dappengine.version=SOME-VERSION
+     $ mvn clean package appengine:deploy -Dappengine.appId=YOUR-PROJECT-ID -Dappengine.version=SOME-VERSION
```
Comment (lesv) [inline appengine-java8/images/README.md]: And why don' t you ask for : -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/oauth2/README.md [hunk 5]
```diff
@@ -21,10 +21,10 @@ Client ID from the prior step.
 ## Running locally
 NOTE: The app can be run locally, but the Oauth2 APIs do not work with the development server.
 
-    $ mvn appengine:devserver
+    $ mvn appengine:run
 
 ## Deploying
-    $ mvn appengine:update -Dappengine.appId=YOUR-PROJECT-ID -Dappengine.version=SOME-VERSION
+    $ mvn package appengine:deploy -Dappengine.appId=YOUR-PROJECT-ID -Dappengine.version=SOME-VERSION
```
Comment (lesv) [inline appengine-java8/oauth2/README.md]: Wow - too long

File: appengine-java8/pubsub/README.md [hunk 6]
```diff
@@ -101,7 +101,7 @@ Update the environment variables `PUBSUB_TOPIC` and `PUBSUB_VERIFICATION_TOKEN`
 then:
 
 ```
-   mvn appengine:deploy
+   mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/pubsub/README.md]: Let's strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/requests/README.md [hunk 7]
```diff
@@ -30,5 +30,4 @@ In the following command, replace YOUR-PROJECT-ID with your
 [Google Cloud Project ID](https://developers.google.com/console/help/new/#projectnumber)
 and SOME-VERSION with a valid version number.
 
-    mvn appengine:deploy
-
+    mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/requests/README.md]: Let's strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/requests/pom.xml [hunk 8]
```diff
@@ -15,113 +15,106 @@ Copyright 2016 Google Inc.
...
-        <artifactId>shared-configuration</artifactId>
+  <!--
+    The parent pom defines common style checks and testing strategies for our samples.
+    Removing or replacing it should not affect the execution of the samples in anyway.
+  -->
+  <?xml version="1.0" encoding="UTF-8"?>
+  <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
     <version>1.0.11</version>
```
Comment (lesv) [inline appengine-java8/requests/pom.xml]: 1.0.12

File: appengine-java8/requests/pom.xml [hunk 9]
```diff
@@ -15,113 +15,106 @@ Copyright 2016 Google Inc.
...
+    <dependency>
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-tools-sdk</artifactId>
+      <version>1.9.78</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.truth</groupId>
+      <artifactId>truth</artifactId>
+      <version>1.0</version>
```
Comment (lesv) [inline appengine-java8/requests/pom.xml]: 1.0.1

File: appengine-java8/requests/pom.xml [hunk 10]
```diff
@@ -15,113 +15,106 @@ Copyright 2016 Google Inc.
...
+  </parent>
+  <properties>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+  </properties>
+  <dependencies>
+    <dependency>
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-maven-plugin</artifactId>
+      <version>1.9.77</version>
```
Comment (lesv) [inline appengine-java8/requests/pom.xml]: 1.9.78 (or latest)

File: appengine-java8/sendgrid/README.md [hunk 11]
```diff
@@ -51,7 +51,7 @@ in a web browser.
 Deploy your application to App Engine standard with the following command:
 
   ```shell
-  mvn appengine:deploy
+  mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/sendgrid/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/spanner/README.md [hunk 12]
```diff
@@ -49,9 +49,8 @@ Note : by default all the spanner example operations run in order, this operatio
 
 ## Deploying
 
-    $ mvn clean appengine:deploy
+    $ mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/spanner/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/taskqueues-pull/README.md [hunk 13]
```diff
@@ -22,5 +22,4 @@ To run this sample locally:
 
 ## Deploying
 
-    mvn appengine:deploy
-
+    mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/taskqueues-pull/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/taskqueues-pull/pom.xml [hunk 14]
```diff
@@ -27,73 +27,68 @@ Copyright 2016 Google Inc.
...
       Removing or replacing it should not affect the execution of the samples in anyway.
     -->
-    <parent>
-        <groupId>com.google.cloud.samples</groupId>
-        <artifactId>shared-configuration</artifactId>
+  <?xml version="1.0" encoding="UTF-8"?>
+  <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
     <version>1.0.11</version>
```
Comment (lesv) [inline appengine-java8/taskqueues-pull/pom.xml]: 1.0.12

File: appengine-java8/taskqueues-push/README.md [hunk 15]
```diff
@@ -24,5 +24,4 @@ Go to the site `localhost:8080` to add elements to the queue.  They will appear
 
 ## Deploying
 
-    mvn appengine:deploy
-
+    mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/taskqueues-push/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/tasks/app/README.md [hunk 16]
```diff
@@ -40,5 +40,5 @@ Second, [Using Maven and the App Engine Plugin](https://cloud.google.com/appengi
 deploy to Google App Engine:
 
 ```
-mvn appengine:deploy
+mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/tasks/app/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/tasks/quickstart/README.md [hunk 17]
```diff
@@ -39,7 +39,7 @@ version unless configured to do otherwise.
 & [Maven Plugin Goals and Parameters](https://cloud.google.com/appengine/docs/flexible/java/maven-reference)
 
 ```
-mvn appengine:deploy
+mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/tasks/quickstart/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/translate-pubsub/README.md [hunk 18]
```diff
@@ -88,7 +88,7 @@ Update the environment variables `PUBSUB_TOPIC` and `PUBSUB_VERIFICATION_TOKEN`
 [`appengine-web.xml`](src/main/webapp/WEB-INF/appengine-web.xml), then:
 
 ```
-   mvn appengine:deploy
+mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/translate-pubsub/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/users/README.md [hunk 19]
```diff
@@ -20,4 +20,4 @@ To run this sample locally:
 In the following command, replace YOUR-PROJECT-ID with your
 [Google Cloud Project ID](https://developers.google.com/console/help/new/#projectnumber).
 
-    $ mvn appengine:deploy
+    $ mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/users/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/websocket-jetty/README.md [hunk 20]
```diff
@@ -51,7 +51,7 @@ For more details on configuring your `app.yaml`, please refer to [this resource]
 The sample application is packaged as a war, and hence will be automatically run using the [Java 8/Jetty 9 with Servlet 3.1 Runtime](https://cloud.google.com/appengine/docs/flexible/java/dev-jetty9).
 
 ```
-mvn appengine:deploy
+mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/websocket-jetty/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: appengine-java8/websocket-jsr356/README.md [hunk 21]
```diff
@@ -50,7 +50,7 @@ For more details on configuring your `app.yaml`, please refer to [this resource]
 The sample application is packaged as a war, and hence will be automatically run using the [Java 8/Jetty 9 with Servlet 3.1 Runtime](https://cloud.google.com/appengine/docs/flexible/java/dev-jetty9).
 
 ```
-  mvn appengine:deploy
+  mvn clean package appengine:deploy -Dapp.deploy.projectId=<your-project-id>
```
Comment (lesv) [inline appengine-java8/websocket-jsr356/README.md]: strike -Dapp.deploy.projectId=<your-project-id>

File: memorystore/redis/pom.xml [hunk 22]
```diff
@@ -70,6 +70,10 @@
         <groupId>com.google.cloud.tools</groupId>
         <artifactId>appengine-maven-plugin</artifactId>
         <version>2.2.0</version>
+        <configuration>
+          <projectId>YOUR_PROJECT_ID</projectId>
```
Comment (lesv) [inline memorystore/redis/pom.xml]: something like GCLOUD ?

File: storage/xml-api/serviceaccount-appengine-sample/pom.xml [hunk 23]
```diff
@@ -82,17 +82,19 @@
...
+        "mvn package appengine:deploy" to upload to GAE.
       -->
       <plugin>
-        <groupId>com.google.appengine</groupId>
+        <groupId>com.google.cloud.tools</groupId>
         <artifactId>appengine-maven-plugin</artifactId>
-        <version>1.9.77</version>
+        <version>2.2.0</version>
         <configuration>
+          <projectId>YOUR_PROJECT_ID</projectId>
```
Comment (lesv) [inline storage/xml-api/serviceaccount-appengine-sample/pom.xml]: GCLOUD ??

File: unittests/pom.xml [hunk 24]
```diff
@@ -14,16 +14,14 @@
     <parent>
         <groupId>com.google.cloud.samples</groupId>
         <artifactId>shared-configuration</artifactId>
-    <version>1.0.11</version>
+        <version>1.0.11</version>
```
Comment (lesv) [inline unittests/pom.xml]: 1.0.12

File: unittests/pom.xml [hunk 25]
```diff
@@ -96,10 +94,14 @@
...
             <plugin>
-                <groupId>com.google.appengine</groupId>
-                <artifactId>appengine-maven-plugin</artifactId>
-                <version>$1.9.60</version>
-            </plugin>
+              <groupId>com.google.cloud.tools</groupId>
+              <artifactId>appengine-maven-plugin</artifactId>
+              <version>2.2.0</version>
+              <configuration>
+                <projectId>YOUR_PROJECT_ID</projectId>
```
Comment (lesv) [inline unittests/pom.xml]: GCLOUD?

File: appengine-java8/bigtable/README.md (hunk 2 above)
Comment (averikitsch) [reply appengine-java8/bigtable/README.md]: It is set using the bigtable project Id because why duplicate.

**Comment by averikitsch** [comment]
going to fix these in another PR. @loosebazooka please advise your team in the future to make these updates when pushing new plugin versions.

**Comment by loosebazooka** [comment]
okay I can add it to our docs. Is it just this repo?

**Comment by averikitsch** [comment]
As well as getting-started-java. Thanks!

**Comment by loosebazooka** [comment]
Is there some kind of robot keeping track of this too?

**Comment by averikitsch** [comment]
We have Renovate bot that will make a PR to auto-update the dependency.

**Pull request closed without merging**